apps/api/src/campaign-report/campaign-report.service.ts

```diff
@@ -0,0 +1,209 @@
+      },
+      select: {
+        id: true,
+        startDate: true,
+        endDate: true,
+        description: true
+      }
+    })
+  }
+
+  async getReport(reportId: string, includeDeleted = false): Promise<GetReportDto | null> {
+    const report = this.prisma.campaignReport.findFirst({
+      where: { 
+        id: reportId,
+        isDeleted: includeDeleted,
+      },
+      select: {
+        files: {
+          where: {
+            isDeleted: false
+          }
+        }
+      }
+    })
+
+    if (report === null) {
+      return null
+    }
+
+    const files = await report.files()
+    const photos = files.filter(file => file.type === CampaignReportFileType.photo)
+    const documents = files.filter(file => file.type === CampaignReportFileType.document)
+  
+
+    return {
+      ...report,
+      photos,
+      documents
+    }
+  }
+
+  async createReport(
+    createReportDto: CreateReportDto,
+    campaignId: string,
+    userId: string,
+    photos: Express.Multer.File[],
+    documents: Express.Multer.File[],
+  ): Promise<string> {
+    const createdReport = await this.prisma.campaignReport.create({
+      data: {
+        ...createReportDto.toEntity(campaignId, userId),
+      },
+    })
+
+    await this.uploadFiles(campaignId, createdReport.id, userId, photos, documents)
+
+    return Promise.resolve(createdReport.description)
+  }
+
+  async updateReport(
+    reportId: string,
+    userId: string,
+    updateReportDto: UpdateReportDto,
+    newPhotos: Express.Multer.File[],
+    newDocuments: Express.Multer.File[],) {
+    const report = await this.prisma.campaignReport.findUnique({
+      where: { id: reportId }
+    })
+
+    if (!report) {
+      return undefined
+    }
+
+    // Update basic report data
+    await this.prisma.campaignReport.update({
+      where: { id: reportId },
+      data: {
+        description: updateReportDto.description,
+        startDate: updateReportDto.startDate,
+        endDate: updateReportDto.endDate,
+        totalFunds: updateReportDto.totalFunds,
+        fundsForPeriod: updateReportDto.fundsForPeriod,
+        spentFundsForPeriod: updateReportDto.spentFundsForPeriod,
+        goals: updateReportDto.goals,
+        nextSteps: updateReportDto.nextSteps,
+        additionalInfo: updateReportDto.additionalInfo
+      }
+    })
+
+    // Mark files as deleted
+    await Promise.all(updateReportDto.daletedFileIds.map(deletedFileId => this.softDeleteFile(deletedFileId)))
+
+    // Upload new files
+    await this.uploadFiles(report.campaignId, reportId, userId, newPhotos, newDocuments)
+  }
+
+  async softDeleteReport(reportId: string): Promise<CampaignReport> {
+    return this.prisma.campaignReport.update({
+      where: {
+        id: reportId
+      },
+      data: {
+        isDeleted: { set: true }
+      }
+    })
+  }
+
+  private async softDeleteFile(fileId: string): Promise<CampaignReportFile> {
+    return this.prisma.campaignReportFile.update({
```

Comment:
when do we actually delete the files?

Reply:
We don't for the moment as per [this reply](https://github.com/podkrepi-bg/frontend/issues/1093#issuecomment-1319845075) from @igoychev. We could implement the actual deletion in several different ways, but I'd propose to leave this for a future task if it's not absolutely required for the initial version of the feature.